Fmaraev/Hometask_2/stepik_org.py

```diff
@@ -0,0 +1,245 @@
+В приведенном ниже примере среднее арифметическое считается
+для чисел на отрезке [−5;12]. Всего чисел, делящихся на 3,
+на этом отрезке 6: −3,0,3,6,9,12. Их среднее арифметическое равно 4.5
+На вход программе подаются интервалы, внутри которых всегда есть
+хотя бы одно число, которое делится на 3."""
+
+a = int(input('Введите число a'))
+b = int(input('Введите число b(>a)'))
+s = 0
+n = 0
+while a % 3 != 0:
+    a += 1
+for i in range(a, b + 1, 3):
+    s += i
+    n += 1
+print('Среднее арифметическое чисел от a до b, делящихся на 3 ', s / n)
+
+"""Узнав, что ДНК не является случайной строкой, только что поступившие
+в Институт биоинформатики студенты группы информатиков предложили
+использовать алгоритм сжатия, который сжимает повторяющиеся символы в строке.
+Кодирование осуществляется следующим образом:
+s = 'aaaabbсaa' преобразуется в 'a4b2с1a2', то есть группы одинаковых символов
+исходной строки заменяются на этот символ и количество его повторений в этой
+позиции строки.
+Напишите программу, которая считывает строку, кодирует её предложенным
+алгоритмом и выводит закодированную последовательность на стандартный вывод.
+Кодирование должно учитывать регистр символов."""
+
+s = input('Введите строку ДНК')
+i = 0
+n = 1
+st = ''
+while i < (len(s) - 1):
+    if s[i] == s[i + 1]:
+        n += 1
+    else:
+        st = st + s[i] + str(n)
+        n = 1
+    i += 1
+print(st + s[i] + str(n))
+
+"""Напишите программу, на вход которой подается одна строка с целыми
+числами. Программа должна вывести сумму этих чисел.
+Используйте метод split строки. """
+
+a = [int(i) for i in input('Введите строку из чисел').split()]
+print(sum(a))
+
+"""Напишите программу, на вход которой подаётся список чисел одной
+строкой. Программа должна для каждого элемента этого списка вывести
+сумму двух его соседей. Для элементов списка, являющихся крайними,
+одним из соседей считается элемент, находящий на противоположном
+конце этого списка. Например, если на вход подаётся список "1 3 5 6 10",
+то на выход ожидается список "13 6 9 15 7" (без кавычек).
+Если на вход пришло только одно число, надо вывести его же.
+Вывод должен содержать одну строку с числами нового списка,
+разделёнными пробелом."""
+
+a = [int(i) for i in input('Введите список из чисел').split()]
+b = []
+if len(a) != 1:
+    for i in range(0, len(a)):
+        b.append(a[(i - 1) % len(a)] + a[(i + 1) % len(a)])
+else:
+    b.append(a[0])
+print(*b, sep=" ")
+
+"""Напишите программу, которая выводит часть последовательности
+1 2 2 3 3 3 4 4 4 4 5 5 5 5 5 ... (число повторяется столько раз,
+чему равно). На вход программе передаётся неотрицательное целое
+число n — столько элементов последовательности должна отобразить
+программа. На выходе ожидается последовательность чисел, записанных
+через пробел в одну строку.
+Например, если n = 7, то программа должна вывести 1 2 2 3 3 3 4."""
+
+a = int(input('Введите число'))
+c = []
+for i in range(1, a + 1):
```

Comment:
it's not optimal solutions( for big numbers we generate tons of chars but print not very big part of them (for n=10 you generate approximately 55 values but print only 10 of them) but it's acceptable